src/ChatWindow.test.js

```diff
@@ -293,13 +301,21 @@ const testOnUserRequestToAlexaSubmitHandling = (
     const finalState = chatWindowInstance.state;
     const finalMessages = finalState.messages;
 
-    expect(finalMessages.length).toBe(numberOfMessagesAlreadyInState + 2);
-    expect(finalMessages[finalMessages.length - 2]).toEqual(
-      expectedUserMessage
-    );
-    expect(finalMessages[finalMessages.length - 1]).toEqual(
-      expectedAlexaResponse
+    // We should have added userMessage and all of Alexa's responses.
+    const numberOfAlexaResponses = expectedAlexaResponses.size;
+    const numberOfNewMessagesInState = 1 + numberOfAlexaResponses;
+
+    expect(finalMessages.length).toBe(
+      numberOfMessagesAlreadyInState + numberOfNewMessagesInState
     );
+    expect(
```

Comment:
can you do this instead?
expect(finalMessages.get(numberOfMessagesAlreadyInState + 1)).toEqual(expectedUserMessage)
expect(finalMessages.slice(numberOfMessagesAlreadyInState + 1)).toEqual(expectedAlexaResponses)

Reply:
Okay.